### DISCUSSION TITLE
[已解决]请问使用VPS如何编译windows 64位的xray.exe

### DISCUSSION THREAD

**Discussion by chika0801** [XTLS/Xray-core#1705] [Q&A] [answered]
我参考某网友在一个ISS回复的，已经会编译linux amd64的xray

使用方法如
https://github.com/chika0801/Xray-examples/blob/main/VLESS-XTLS-uTLS-REALITY/compile_Xray-core.md

我参照官方readme，我在debian11的VPS上，使用下面命令

`CGO_ENABLED=0 go build -o xray.exe -trimpath -ldflags "-s -w -buildid=" ./main`

得到的这个.exe在win11 64位下不能运行。

请问正确的命令是什么？

**Comment by cross-hello** [comment, marked as answer]
What is your error message?

**Reply by chika0801** [reply]
> What is your error message?

![1](https://user-images.githubusercontent.com/88967758/221120732-ebe2474c-86fd-4baa-8690-ffc98a9fd55c.jpg)

**Reply by cross-hello** [reply]
Have a try
```bat
go env  -w GOARCH=amd64
go build -o xray.exe -trimpath -ldflags "-s -w -buildid=" ./main
```

**Reply by cross-hello** [reply]
Oh, sorry. 
If you compile in Linux, you should specify golang system and architecture:
`GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -o xray.exe -trimpath -ldflags "-s -w -buildid=" ./main`

**Reply by chika0801** [reply]
> ```batchfile
> go build -o xray.exe -trimpath -ldflags "-s -w -buildid=" ./main
> ```

我试了，失败了。谢谢你。

**Reply by chika0801** [reply]
> GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -o xray.exe -trimpath -ldflags "-s -w -buildid=" ./main

谢谢你，成功了。

**Comment by cross-hello** [comment]
So the interesting point here is Windows 11 don't compatible with Windows 3.1.

(in fact you could install a dual boot windows 11 32-bits version to run 16 bits applications)
